pkg/gatewayserver/io/io.go

```diff
@@ -282,16 +327,23 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 	var fp *frequencyplans.FrequencyPlan
 	fpID := request.GetFrequencyPlanID()
 	if fpID != "" {
-		fp = c.fps[fpID]
+		fp = c.gatewayFPs[fpID]
 		if fp == nil {
-			return 0, errFrequencyPlanNotConfigured.WithAttributes("id", request.FrequencyPlanID)
+			// The requested frequency plan is not configured for the gateway. Load the plan and enforce that it's in the same band.
+			fp, err = c.fps.GetByID(fpID)
+			if err != nil {
+				return 0, errFrequencyPlanNotConfigured.WithCause(err).WithAttributes("id", request.FrequencyPlanID)
+			}
+			if fp.BandID != c.bandID {
+				return 0, errFrequencyPlansNotFromSameBand
+			}
```

Comment:
@htdvisser @KrishnaIyer FYI we're loading the frequency plan in the hot path if the NS's frequency plan doesn't match that of the gateway.

We have these circumstances with `EU_863_870` vs `EU_863_870_TTN` where we do need to check band. We might need to tune frequency plan store caching (it's already enabled, but implement background refresh so they're immediately available from memory).